### ISSUE TITLE
socketOptions sends an integer in yaml which is transformed into a string

### ISSUE THREAD

**Issue by antonioberben** [solo-io/gloo#5081]
When installing a proxy with `socketOptions`, the `level` attribute is an `integer`.
However, in the server side, Gateway application changes it somehow to string to be used. 
That change produces an error reported as follows:
```
{"level":"error","ts":1627466232.612716,"logger":"gateway.v1.event_loop.gateway","caller":"syncer/setup_syncer.go:296","msg":"error: event_loop: 1 error occurred:\n\t* reconciling resource gateway-proxy: updating kube resource gateway-proxy:2508 (want 2508): Proxy.gloo.solo.io \"gateway-proxy\" is invalid: [spec.listeners.options.socketOptions.intValue: Invalid value: \"string\": spec.listeners.options.socketOptions.intValue in body must be of type integer: \"string\", spec.listeners.options.socketOptions.level: Invalid value: \"string\": spec.listeners.options.socketOptions.level in body must be of type integer: \"string\", spec.listeners.options.socketOptions.name: Invalid value: \"string\": spec.listeners.options.socketOptions.name in body must be of type integer: \"string\"]\n\n","version":"1.8.2","stacktrace":"github.com/solo-io/gloo/projects/gateway/pkg/syncer.RunGateway.func1\n\t/workspace/gloo/projects/gateway/pkg/syncer/setup_syncer.go:296"}
```

# How to reproduce:

1. Install gloo `1.8.1`  through helm with following values:
```
license_key: XXXXXX
create_license_secret: true 
global:
  extensions:
    extAuth:
      enabled: true
      envoySidecar: false
  glooStats:
    enabled: true
gloo:
  crds:
    create: true
  discovery:
    enabled: true
  gatewayProxies:
    gatewayProxy:
      gatewaySettings:
        options:
          socketOptions:
            - description: "enable keep-alive"
              level: "1"
              name: 9
              intValue: 1
              state: STATE_LISTENING
```
2. Once the installation is completed, check te gateway logs:
```
kubectl logs -n gloo-system -l gloo=gateway
```

You will see the error described at the beginning of this issue

Thank you

**Comment by sam-heilbron** [comment]
Hi @antonioberben, did you updgrade your CRDs? This should have been resolved by: https://github.com/solo-io/gloo/pull/5055/ which was created from as a result of this issue: https://github.com/solo-io/gloo/issues/5043

**Comment by antonioberben** [comment]
@sam-heilbron , you are right. It is solved in `1.8.1`

For whoever might concern. As of today:
**It is important to make sure to upgrade also the CRDs in each upgrade**
